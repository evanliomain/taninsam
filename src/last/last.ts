/**
 * @module array=>any
 */
/**
 * Extract the last element of an array. An empty array has an undefined last element
 * @return the function to apply on the TBD to do something
 * @example
 * ```
 * last<number>()([1, 2, 3, 4, 5]) // 5
 * ```
 * @example Using the chain
 * ```
 * chain([1, 2, 3, 4, 5])
 *   .chain(last<number>())
 *   .value() // 5
 * ```
 */
export function last<T>(): (element: ReadonlyArray<T>) => T | undefined {
  return (array: ReadonlyArray<T>): T | undefined => {
    if (0 === array.length) {
      return undefined;
    }
    return array[array.length - 1];
  };
}
