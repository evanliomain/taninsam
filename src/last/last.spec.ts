import { last } from './last';

describe('last function', () => {
  const input = [1, 2, 3, 4, 5];

  test('is a pure function with array', () => {
    last()(input);
    expect(input).toMatchSnapshot();
  });

  test('[1, 2, 3, 4, 5] |> last === 5', () => {
    expect(last()(input)).toMatchSnapshot();
  });

  test('[] |> last === undefined', () => {
    expect(last()([])).toMatchSnapshot();
  });
});
