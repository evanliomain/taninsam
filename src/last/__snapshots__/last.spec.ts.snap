// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`last function > [] |> last === undefined 1`] = `undefined`;

exports[`last function > [1, 2, 3, 4, 5] |> last === 5 1`] = `5`;

exports[`last function > is a pure function with array 1`] = `
[
  1,
  2,
  3,
  4,
  5,
]
`;
